src/components/scenes/SwapCreateScene.tsx

```diff
@@ -279,41 +277,32 @@ export const CryptoExchangeScene = (props: Props) => {
     showWalletListModal('to')
   })
 
-  const handleFromFocusWallet = useHandler(() => {
-    setState({
-      ...state,
-      whichWalletFocus: 'from'
-    })
-  })
-
-  const handleToFocusWallet = useHandler(() => {
-    setState({
-      ...state,
-      whichWalletFocus: 'to'
-    })
-  })
-
   const handleFromAmountChange = useHandler((amounts: ExchangedFlipInputAmounts) => {
     setState({
       ...state,
-      fromAmountNative: amounts.nativeAmount
+      nativeAmount: amounts.nativeAmount,
+      nativeAmountFor: 'from'
     })
+    // Clear other input's amount:
+    toInputRef.current?.setAmount('crypto', '0')
   })
 
   const handleToAmountChange = useHandler((amounts: ExchangedFlipInputAmounts) => {
     setState({
       ...state,
-      toAmountNative: amounts.nativeAmount
+      nativeAmount: amounts.nativeAmount,
+      nativeAmountFor: 'to'
     })
+    // Clear other input's amount:
+    fromInputRef.current?.setAmount('crypto', '0')
   })
 
   //
   // Render
   //
 
   const renderButton = () => {
-    const primaryNativeAmount = state.whichWalletFocus === 'from' ? state.fromAmountNative : state.toAmountNative
-    const showNext = fromCurrencyCode !== '' && toCurrencyCode !== '' && !!parseFloat(primaryNativeAmount)
+    const showNext = fromCurrencyCode !== '' && toCurrencyCode !== '' && !!parseFloat(state.nativeAmount)
```

Comment:
We should probably do `> 0` instead of `!!`. The `> 0` will return false for 0 and NaN, which are the two cases we care about.